src/functional/ffront/past_to_itir.py

```diff
@@ -205,6 +217,39 @@ def _construct_itir_domain_arg(
 
         return itir.FunCall(fun=itir.SymRef(id=domain_builtin), args=domain_args)
 
+    def _construct_itir_initialized_domain_arg(
+        self,
+        dim_i: int,
+        dim: Dimension,
+        node_domain: past.Dict,
+    ) -> tuple[itir.Literal, itir.Literal]:
+        try:
+            if node_domain.keys_[dim_i].type.dim == dim:
+                domain_bounds = []
+                for domain_i, domain_input in enumerate(node_domain.values_[dim_i].elts):
```

Comment:
I would remove the for loop as we already know it has only 2 values and do something like 
```
assert len(node_domain.values_[dim_i].elts) == 2 
start, end = node_domain.values_[dim_i].elts
```
(assert here is fine as this should never trigger because we did a check in the frontend already)